testsuite/tests/weak-ephe-final/weaklifetime_par.ml

```diff
@@ -0,0 +1,91 @@
+(* TEST
+*)
+
+Random.init 12345;;
+
+let size = 1000;;
+let num_domains = 4;;
+let random_state = Domain.DLS.new_key Random.State.make_self_init
+
+type block = int array;;
+
+type objdata =
+  | Present of block
+  | Absent of int  (* GC count at time of erase *)
+;;
+
+type bunch = {
+  objs : objdata array;
+  wp : block Weak.t;
+};;
+
+let data =
+  Array.init size (fun i ->
+    let n = 1 + Random.int size in
+    {
+      objs = Array.make n (Absent 0);
+      wp = Weak.create n;
```

Comment:
Right now this will create all the weak pointers in the initial main domain and so will not undergo a domain lifecycle event. 
Are we able to change the initialization of the data to be done per domain?
Then each domain would have created weak pointers and experienced lifecycle events with that.